### PR TITLE
AArch64: map kernel pages with VM_PROT_NONE without read permissions.

### PR DESCRIPTION
**Pull request by pj1031999**
Map kernel pages with `VM_PROT_NONE` without read permissions.

It fixes pmap_kenter and rmbits from #932.

### CODE REVIEW

File: sys/aarch64/pmap.c [hunk 1]
```diff
@@ -43,7 +43,7 @@ static const pte_t pte_default =
 static const pte_t pte_noexec = ATTR_XN | ATTR_SW_NOEXEC;
 
 static const pte_t vm_prot_map[] = {
-  [VM_PROT_NONE] = ATTR_XN | pte_noexec | pte_default,
+  [VM_PROT_NONE] = L3_PAGE | ATTR_SH(ATTR_SH_IS) | pte_noexec,
```
Comment (cahirwpz) [inline sys/aarch64/pmap.c:46]: Please remove `ATTR_SH`, `ATTR_AP` and `ATTR_S2_S2AP*`. These macro does not introduce any important concept and are cluttering our code.

If `pte_common` is `L3_PAGE | ATTR_SH(ATTR_SH_IS)` then should we set `ATTR_DBM` & `ATTR_AF` for all other cases? 

Could you add docstrings for `ATTR_DBM` & `ATTR_AF`? I do not remember what the mean...

Comment (pj1031999) [reply sys/aarch64/pmap.c]: I removed some macros and added comments about `DBM` and `AF` bits - please check if it's enough.

`DBM` means that page is in dirty state, `AF` means that page is valid.
I think that we should set `AF` flag for every usable kernel page - we already do that for _mips_.
For `DBM` flag you are right. I removed that flag from RO pages but I'm not sure about kernel pages. For _mips_ we set that flag for every writable kernel page but - please correct me if I'm wrong - dirty flag on _mips_ also means write access for page. From my point of view we can safely remove `DBM` flag from `vm_prot_map` and set it only from `pmap_emulate_bits` but still we should set it manually for initial page table in `boot.c`.

Both of them _could_ be managed by hardware - see `HD` and `HA` bits in TCR_EL (Translation Control Register) but we don't use that feature now.

Comment (cahirwpz) [reply sys/aarch64/pmap.c]: I updated DBM & AF description – I think its clearer now.

To follow _MIPS_ behaviour `DBM` and `AF` flags **must not** be set by default for user pages. For read-only kernel pages we need to set `AF` for read-write `DBM` as well.

Comment (pj1031999) [reply sys/aarch64/pmap.c]: As we discussed offline:
- I reverted DBM description change
- `pte_default` was renamed into `pte_common`
- `AF` now is explicit
- `AF` and `DBM` wasn't set for user pages exactly like in _mips_ - see `pmap_enter`

**Review by cahirwpz — APPROVED** [review]
LGTM

**Pull request merged**